modules/openapi-generator/src/main/resources/swift5/APIs.mustache

```diff
@@ -56,8 +56,11 @@ import Vapor
             headers[header] = value
         }
     }
-
-    {{#nonPublicApi}}internal{{/nonPublicApi}}{{^nonPublicApi}}open{{/nonPublicApi}} func execute(_ apiResponseQueue: DispatchQueue = {{projectName}}API.apiResponseQueue, _ completion: @escaping (_ result: Swift.Result<Response<T>, ErrorResponse>) -> Void) { }
+    {{#useURLSession}}{{^useVapor}}{{^usePromiseKit}}{{^useResult}}{{^useAsyncAwait}}
```

Comment:
@DavidHorvath do we need those conditions?
I don't we should return the `URLSessionDataTask` in every configuration, and then in the `api.mustache`we add the conditions, case by case.

Reply:
I've only added this condition because I only knew how I need to cancel request for URLSession with Default, Combine or RxSwift configurations and I didn't want to update the others 🙂 

But with your comments and links above, I have a direction to do it.
So if I update the other configurations as well, we won't need this

Reply:
Sure, and you did great, because honestly in the past I looked at this big task and felt a bit discouraged.
If you need any help please let me know 👍